apps/web/src/services/asset-inventory/collector/collector-main/CollectorPage.vue

```diff
@@ -166,7 +169,16 @@ onUnmounted(() => {
         store.dispatch('reference/plugin/load'),
         store.dispatch('reference/provider/load'),
     ]);
-    await initCollectorList();
+    await collectorPageStore.$reset();
+    if (Object.keys(query).length > 0) {
+        const { provider } = query;
+        if (provider === 'all') {
+            await initCollectorList();
+        }
+        await handleSelectedProvider(provider as string);
```

Comment:
nitpick: Actually I think calling directly an event handler devastates readability and maintainability. 
But I totally understand why you did use this handler. 🥲 It's really cumbersome to write a separate function because it has only one simple line!
Let's leave it and let the maintainer separate it if he or she needs to differentiate two functions: an event handler and an initializer.